src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -79,48 +80,62 @@ public ActivityLogEntry(String servlet, String params, String link){
      */
     public ActivityLogEntry(AppLogLine appLog){
         time = appLog.getTimeUsec() / 1000;
-        String[] tokens = appLog.getLogMessage().split("\\|\\|\\|", -1);
-        
-        //TEAMMATESLOG|||SERVLET_NAME|||ACTION|||TO_SHOW|||ROLE|||NAME|||GOOGLE_ID|||EMAIL|||MESSAGE(IN HTML)|||URL|||ID|||TIME_TAKEN
-        try{
-            servletName = tokens[1];
-            action = tokens[2];
-            toShow = (tokens[3].equals("true") ? true : false);
-            role = tokens[4];
-            name = tokens[5];
-            googleId = tokens[6];
-            email = tokens[7];
-            message = tokens[8];
-            url = tokens[9];
-            if (tokens.length >= 11) {
-                if (tokens[10].contains(googleId)) {
-                    timeTaken = tokens.length == 12 ? Long.parseLong(tokens[11].trim()) : null;
-                    id = tokens[10];
-                } else {
-                    timeTaken = Long.parseLong(tokens[10].trim());                    
-                }
-            }
-
-            keyStringsToHighlight = null;
+        
+        try {
+            String[] tokens = appLog.getLogMessage().split("\\|\\|\\|", -1);
+            initUsingAppLogMessage(tokens);
         } catch (ArrayIndexOutOfBoundsException e){
-            
-            servletName = "Unknown";
-            action = "Unknown";
-            role = "Unknown";
-            name = "Unknown";
-            googleId = "Unknown";
-            email = "Unknown";
-            toShow = true;
-            message = "<span class=\"text-danger\">Error. Problem parsing log message from the server.</span><br>"
-                    + "System Error: " + e.getMessage() + "<br>" + appLog.getLogMessage();
-            url = "Unknown";
-            id = "Unknown" + formatTimeForId(new Date(time));
-            timeTaken = null;
-            keyStringsToHighlight = null;
+            initAsFailure(appLog, e);
         }
         
+        keyStringsToHighlight = null;
         logInfoAsHtml = getLogInfoForTableRowAsHtml();
     }
+
+
+    private void initUsingAppLogMessage(String[] tokens) {
+        servletName = tokens[1];
+        action = tokens[2];
+        toShow = Boolean.parseBoolean(tokens[3]);
+        role = tokens[4];
+        name = tokens[5];
+        googleId = tokens[6];            
+        email = tokens[7];
+        message = tokens[8];
+        url = tokens[9];
+        
+        boolean isLogWithoutTimeTakenAndId = tokens.length < 11;
+        if (!isLogWithoutTimeTakenAndId) {
+            boolean isOldLog = !(tokens[10].contains(googleId) || tokens[10].contains("%"));
```

Comment:
In this case, the possible values are either the timestamp, or the googleId (or "unknown"/"unregistered"), so false positives shouldn't happen